core/libp2p/basic/varint_reader.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_VARINT_READER_HPP
+#define KAGOME_VARINT_READER_HPP
+
+#include <functional>
+#include <memory>
+#include <optional>
+
+#include "common/buffer.hpp"
+#include "libp2p/basic/readwriter.hpp"
+#include "libp2p/multi/uvarint.hpp"
+
+namespace libp2p::basic {
+  class VarintReader {
+   public:
+    static constexpr uint8_t kMaximumVarintLength = 9;  // taken from Go
+
+    /**
+     * Read a varint from the connection
+     * @param conn to be read from
+     * @param cb to be called, when a varint or maximum bytes from the
+     * connection are read
+     */
+    static void readVarint(
```

Comment:
why is it a class with static member? 

why not make it part of message reader?

Reply:
I think it makes sense to separate logic of reading message and reading varint

Reply:
Yes, I wanted to do it, because `VarintReader` can be reused in other classes, which want to read varints